lib/subscription/runner.ex

```diff
@@ -0,0 +1,46 @@
+defmodule AshGraphql.Subscription.Runner do
+  alias Absinthe.Pipeline.BatchResolver
+
+  require Logger
+
+  def run_docset(pubsub, docs_and_topics, notification) do
+    for {topic, key_strategy, doc} <- docs_and_topics do
+      try do
+        pipeline =
+          Absinthe.Subscription.Local.pipeline(doc, notification)
+
+        {:ok, %{result: data}, _} = Absinthe.Pipeline.run(doc.source, pipeline)
+
+        Logger.debug("""
+        Absinthe Subscription Publication
+        Field Topic: #{inspect(key_strategy)}
+        Subscription id: #{inspect(topic)}
+        Data: #{inspect(data)}
+        """)
+
+        case should_send?(data) do
+          false ->
+            :ok
+
+          true ->
+            :ok = pubsub.publish_subscription(topic, data)
+        end
+      rescue
+        e ->
+          BatchResolver.pipeline_error(e, __STACKTRACE__)
+      end
+    end
+  end
+
+  defp should_send?(%{errors: errors}) do
+    # if the user is not allowed to see the data or the query didn't
+    # return any data we do not send the error to the client
+    # because it would just expose unnecessary information
+    # and the user can not really do anything usefull with it
+    not (errors
+         |> List.wrap()
+         |> Enum.any?(fn error -> Map.get(error, :code) in ["forbidden", "not_found"] end))
```

Comment:
Maybe also just for safety include `nil` here. i.e if we don't know what kind of error it was.